### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by VasilMitovClouway**
Task 3 is ready for review.

### CODE REVIEW

File: src/com/clouway/adapter/PersistentStockRepository.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import com.clouway.core.Stock;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class PersistentStockRepository implements StockRepository {
```
Comment (freekman) [inline src/com/clouway/adapter/PersistentStockRepository.java]: missing test cases

File: src/com/clouway/core/Provider.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.clouway.core;
+
```
Comment (freekman) [inline src/com/clouway/core/Provider.java]: .

File: src/com/clouway/adapter/PersistentStockHistoryRepository.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class PersistentStockHistoryRepository implements StockHistoryRepository {
+  private Provider<Connection> provider;
+  private Integer pageSize;
```
Comment (freekman) [inline src/com/clouway/adapter/PersistentStockHistoryRepository.java]: ?

File: test/HistoryTest.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class HistoryTest {
```
Comment (freekman) [inline test/HistoryTest.java]: the naming for the Class in not very good

File: src/com/clouway/adapter/PersistentStockHistoryRepository.java (hunk 3 above)
Comment (VasilMitovClouway) [inline src/com/clouway/adapter/PersistentStockHistoryRepository.java]: Leftup code from my last implementation. Removed.

File: test/DisplayingHistoryTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
+import com.clouway.adapter.ConnectionProvider;
```
Comment (freekman) [inline test/DisplayingHistoryTest.java]: the name is still bad

File: test/DisplayingHistoryTest.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  @Before
+  public void fillingUp() {
+    deleteTable("STOCK_HISTORY");
+    deleteTable("STOCK");
+    Stock apple = new Stock("apple", 1.2, 3.5);
+    Stock pear = new Stock("pear", 2.2, 4.6);
+    Stock orange = new Stock("orange", 3.1, 5.1);
+    Stock potato = new Stock("potato", 2.3, 6.5);
+
+
```
Comment (freekman) [inline test/DisplayingHistoryTest.java]: remove the extra empty lines

**Pull request merged**